src/eth/EthereumCdpService.js

```diff
@@ -140,15 +147,22 @@ export default class EthereumCdpService extends PrivateService {
     return this._tubContract().draw(hexCdpId, value);
   }
 
-  async wipeDai(cdpId, amount, unit = DAI) {
-    await this._checkEnoughMkr(cdpId);
-    const hexCdpId = numberToBytes32(cdpId);
-    const value = getCurrency(amount, unit).toEthersBigNumber('wei');
-    await Promise.all([
-      this.get('allowance').requireAllowance(MKR, this._tubContract().address),
-      this.get('allowance').requireAllowance(DAI, this._tubContract().address)
-    ]);
-    return this._tubContract().wipe(hexCdpId, value);
+  wipeDai(cdpId, amount, unit = DAI) {
+    const promise = (async () => {
+      await this._checkEnoughMkr(cdpId);
+      const hexCdpId = numberToBytes32(cdpId);
+      const value = getCurrency(amount, unit).toEthersBigNumber('wei');
+      await Promise.all([
+        this.get('allowance').requireAllowance(
+          MKR,
+          this._tubContract().address
+        ),
+        this.get('allowance').requireAllowance(DAI, this._tubContract().address)
+      ]);
+      return this._tubContract().wipe(hexCdpId, value, { promise });
+    })();
+
+    return promise;
```

Comment:
step 3: make sure it's returned, so that the caller can use it to look up the transactions they want to add event callbacks for

Reply:
I hope that these steps can be wrapped up into a method decorator so that we can end up with something like this:

```js
@trackTransactions
wipeDai(cdpId, amount, unit = DAI, promise) {
  // ...snip
  return this._tubContract().wipe(hexCdpId, value, { promise });
}